src/set_theory/surreal.lean

```diff
@@ -352,10 +352,23 @@ instance : add_semigroup surreal :=
 { add_assoc := add_assoc,
   ..(by apply_instance : has_add surreal) }
 
+theorem zero_add : ∀ (x : surreal), 0 + x = x :=
+by rintro ⟨x, ox⟩; exact quotient.sound (pgame.zero_add_equiv _)
+
+theorem add_zero : ∀ (x : surreal), x + 0 = x :=
+by rintro ⟨x, ox⟩; exact quotient.sound (pgame.add_zero_equiv _)
+
+instance : add_monoid surreal :=
+{ add := add,
+  add_assoc := add_assoc,
+  zero := 0,
+  zero_add := zero_add,
+  add_zero := add_zero }
```

Comment:
```suggestion
{ zero := 0,
  zero_add := zero_add,
  add_zero := add_zero,
  ..(infer_instance : add_semigroup surreal) }
```
I find this style slightly more reassuring, so I don't have to double check that you're really using the same addition!

Reply:
I'll follow your advice below and just define the advanced instances and derive the smaller ones from them.

Reply:
For now, I'll just do it for add_monoid but in the next PR I'll make sure to combine several instances together.